lib/protip/transformers/big_decimal_transformer.rb

```diff
@@ -21,7 +21,7 @@ def to_message(object, field)
               field.subtype.msgclass.new(
                 numerator: rational.numerator,
                 denominator: rational.denominator,
-                precision: object.precs[0], # This is the current precision of the decimal
+                precision: object.try(:precision) || object.precs[0], # This is the current precision of the decimal
```

Comment:
this isn't returning the same value for a basic example like

a = BigDecimal("0.123")
a.precision => 3
a.precs[0] => 9